packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1383,12 +1383,215 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
+///         hasScrollBody: false,
+///         child: Container(
+///           color: Colors.orange[300],
+///           child: Padding(
+///             padding: const EdgeInsets.all(50.0),
+///             child: FlutterLogo(size: 100),
+///           ),
+///         ),
+///       ),
+///     ],
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// * [SliverFillRemaining] will defer to the size of its [child] if the
+///   [precedingScrollExtent] exceeded the length of the viewport's main axis.
+///
+/// {@animation 250 500 https://flutter.github.io/assets-for-api-docs/assets/widgets/sliver_fill_remaining_scrolled_beyond.mp4}
+///
+/// {@tool snippet --template=stateless_widget_scaffold}
+///
+/// In this sample the [SliverFillRemaining] defers to the size of its [child]
+/// because the [precedingScrollExtent] of the [SliverConstraints] has gone
+/// beyond that of the viewport's main axis.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return CustomScrollView(
+///     slivers: <Widget>[
+///       SliverFixedExtentList(
+///         itemExtent: 130.0,
+///         delegate: SliverChildBuilderDelegate(
+///           (BuildContext context, int index) {
+///             return Container(
+///               color: index % 2 == 0
+///                 ? Colors.indigo[200]
+///                 : Colors.orange[200],
+///             );
+///           },
+///           childCount: 5,
+///         ),
+///       ),
+///       SliverFillRemaining(
+///         hasScrollBody: false,
+///         child: Container(
+///           child: Padding(
+///             padding: const EdgeInsets.all(50.0),
+///             child: Icon(
+///               Icons.pan_tool,
+///               size: 60,
+///               color: Colors.blueGrey,
+///             ),
+///           ),
+///         ),
+///       ),
+///     ],
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// * For [ScrollPhysics] that allow overscroll, such as
+///   [BouncingScrollPhysics], setting the [fillOverscroll] flag to true allows
+///   the size of the [child] to _stretch_, filling the overscroll area. It does
+///   this regardless of the path chosen to provide the child's size.
+///
+/// {@animation 250 500 https://flutter.github.io/assets-for-api-docs/assets/widgets/sliver_fill_remaining_fill_overscroll.mp4}
+///
+/// {@tool snippet --template=stateless_widget_scaffold}
+///
+/// In this sample the [SliverFillRemaining]'s child stretches to fill the
+/// overscroll area when [fillOverscroll] is true. This sample also features a
+/// button that is pinned to the bottom of the sliver, regardless of size or
+/// overscroll behavior. Try switching [fillOverscroll] to see the difference.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return CustomScrollView(
+///     // These ScrollPhysics are the default for iOS.
+///     // The fillOverscroll property of SliverFillRemaining is only relevant
+///     // for ScrollPhysics like this.
```

Comment:
Why?  

What should I do if I'm using the platform's default physics and I want to look just right on both Android and iOS?

Reply:
I'm not sure what you mean specifically. Are you talking about pinning the button?

This is just meant to be an example of `fillOverscroll`, not specifically pinning a button. You can pin a button in the examples above that also demonstrate alignment and never use `fillOverscroll`. Other physics don't have an overscroll at all so I thought mentioning its restriction would be a good idea.

I considered pulling on the thread and going further into scroll physics, but it started to feel like scope creep, and off the topic of the widget at hand.

WDYT?